macros/src/main/scala/com/softwaremill/macwire/dependencyLookup/ValuesOfTypeInEnclosingClassFinder.scala

```diff
@@ -27,64 +31,187 @@ private[dependencyLookup] class ValuesOfTypeInEnclosingClassFinder[C <: Context]
    */
   def find(t: Type, implicitValue: Option[Tree] = None): List[Tree] = {
 
+    // Find will look at:
+    //   1. vals or defs
+    //   2. imports
+    //   3. members of vals whose type is annotated with @Module
+    //
+    // A problem arises when a member is found eligible by 2) and 3).
+    // Here is an example:
+    //
+    //    class A
+    //    class B(a: A)
+    //
+    //    @Module
+    //    class M(val a: A)
+    //
+    //    class App(m: M) {
+    //      import m.a
+    //
+    //      lazy val b = wire[B]
+    //    }
+    //
+    // B requires an A, which is found twice:
```

Comment:
I've had the same problem with implicit members (found both because they are a member and because they are implicit). Using a `Set` didn't work there (the trees where not equal, even though they pointed to the same value), so I'm using `PositionUtil.samePosition` - maybe it can be helpful here as well.